base/strings/basic.jl

```diff
@@ -352,31 +352,33 @@ end
 If `i` is in bounds in `s` return the index of the start of the character whose
 encoding code unit `i` is part of. In other words, if `i` is the start of a
 character, return `i`; if `i` is not the start of a character, rewind until the
-start of a character and return that index. If `i` is out of bounds in `s`
-return `i`.
+start of a character and return that index. If `i` is equal to 0 or `ncodeunits(s)+1`
+return `i`. In all other cases throw `BoundsError`.
 
 # Examples
 ```jldoctest
-julia> thisind("αβγdef", -5)
--5
+julia> thisind("α", -1)
```

Comment:
Perhaps start with an example that "works" and leave the `BoundsError `examples to last?

Reply:
moved (I wanted to sow what happens if we increase the index but I agree with your reasoning).